### ISSUE TITLE
Iterating over a list of functions with arity > 1 and calling then in a for in do loop produces incorrect output

### ISSUE THREAD

**Issue by felixschorer** [fable-compiler/Fable#2047]
### Description

Iterating over a list of functions with arity > 1 and calling then in a for in do loop produces incorrect output.

The functions are curried, but are called like uncurried functions.

### Repro code

```f#
let fortyTwo () () =
    42

for someFun in [fortyTwo] do
    printfn "%A" <| someFun () ()
```

[Repl link](https://fable.io/repl/#?code=DYUwLgBAZg9gTmAngFQO4wgCgJRdwXgFgAoCMiAFgCYSTY4IBnGAWxADEBXAOwgEteAbXpI0MALoQAJjBLkIABzgCwUXgCIApAEF1EADwAfJqw488eIA&html=Q&css=Q)

### Expected and actual results

Expected:
```
42
```

Actual:
```
0
```

### Related information

* Fable version: 2.8.4
* Operating system: Window 10 Pro 1903

**Comment by jordwest** [comment]
I'm facing the same issue, it seems to be something inherent in the `for..in` as replacing it with this works fine:

```fsharp
[fortyTwo]
    |> Seq.iter (fun someFun -> printfn "%A" <| someFun () ())
```

It's a sneaky one because it only shows up in this specific case when you have > 1 arguments, and it fails silently if you're expecting it to return a `unit`

**Comment by alfonsogarciacaro** [comment]
This took a while but should be finally fixed in 3.1.3 😸
